Ten samples from the Tensleep Sandstone of the Bighorn Basin in Wyoming were collected and analyzed for porosity (%). The results are shown below under the variable "data."

Part A- Assuming porosity in the formation is approximately normally distributed, compute a 90% confidence interval for both the mean and variance of the porosity of the sandstone formation. 


In [ ]:
import numpy as np #importing Python libraries used for statistical analysis
import scipy as sp
import scipy.stats

In [2]:
data= [13,17,15,23,27,29,18,27,20,24]

def mean_confidence_interval(data, confidence=0.90): #creating a function that returns a 90% confidence interval
    a = 1.0*np.array(data)
    n = len(a)
    m, se = np.mean(a), scipy.stats.sem(a) #calculates mean and standard error
    h = se * sp.stats.t.ppf((1+confidence)/2., n-1)
    return m, m-h, m+h

In [4]:
mean_confidence_interval(data, confidence=0.90) #calling the function

(21.300000000000001, 18.100939271806013, 24.499060728193989)

The mean 90% confidence interval is 18.1< u <24.5 

In [5]:
def var_confidence_interval(data, confidence=0.90): #similarly, creating a function for the variance confidence interval 
    a = 1.0*np.array(data)
    n = len(a)
    v, se = np.var(a), scipy.stats.sem(a) #calculates variance and standard error
    h = se * sp.stats.t.ppf((1+confidence)/2., n-1)
    return v, v-h, v+h

In [6]:
var_confidence_interval(data, confidence=0.90)

(27.410000000000004, 24.210939271806012, 30.609060728193995)

The 90% variance confidence interval is 24.2< var < 30.6

Part B- Use Shapiro-Wilk test to determine whether the porosity data were drawn from a normally distributed population

In [10]:
#Using Python's Scipy library, performing a Shaprio Wilk test is as simple as using the stats.shapiro built in function
#Doing so returns the value of the test statistic, W and the p value

sp.stats.shapiro(data)

(0.9489189386367798, 0.6557270884513855)

Because the test stastistic is larger than the critical value, we can safely conclude this data was drawn from a normal distribution